lib/chef/application/client.rb

```diff
@@ -454,11 +460,12 @@ def interval_run_chef_client
   end
 
   def sleep_then_run_chef_client(sleep_sec)
-    @signal = test_signal
-    unless @signal == IMMEDIATE_RUN_SIGNAL
-      Chef::Log.debug("Sleeping for #{sleep_sec} seconds")
-      interval_sleep(sleep_sec)
-    end
```

Comment:
client client starts, you come into sleep_then_run_chef_client, you haven't had any signals yet because you just started so you go into interval_sleep

in interval_sleep, we hang out and watch the pipe. If something hits it, we're going to drop back out below on 466. One of two things will hit the pipe. An `RECONFIGURE_SIGNAL` (H) will cause us to reconfigure on 468. if it isn't that it is `IMMEDIATE_RUN` (1). In which case, we're about to run anyway, so let's do that (I could add a comment or a line here around 468 to be more clear about that).

Then we run on 471, and we're going to end up looping back around to the top of this method. `test_signal` used to use `interval_sleep` to look to see if anything was sitting in the pipe, but that's redundant because we're just about to call `interval_sleep` anyway, there's nothing happening in between. We don't need a quick check because `interval_sleep` with a longer timeout is still going to return as soon as you send a signal and a character gets dropped into the pipe, or is already there waiting because it was sent in the middle of a previous run.

This is a bit confusing, I tried to improve it which was part of why `test_signal` got renamed to `check_for_signal` but when I realized the above I just avoided it and it looks like I forgot to prune it.